src/utils/getTokenPriceInUSD.ts

```diff
@@ -0,0 +1,175 @@
+import { ethers } from "ethers";
+import { getRpcUrl } from "./getRpcUrl.js";
+import {
+  ChainId,
+  currenciesByNetwork,
+  Currency,
+} from "@hypercerts-org/marketplace-sdk";
+
+export const getTokenPriceInUSD = async (
```

Comment:
this also reflects on the `cheapestOrder` fields resolution, but this needs tests. The business logic seems to check out, but unit tests with happy and unhappy flows are valuable

Reply:
I added tests for `getCheapestOrder` and `getMaxUnitsForSaleInOrders`. I'm not entirely sure what to test in `getTokenPricesInUSD`, except for the line `return Number(formatUnits(roundData * BigInt(100), decimals)) / 100;`, which seems like testing the implementation for formatUnits?